### PR TITLE
Expand and change dialog focus delegation tests

### PR DESCRIPTION
**Pull request by domenic**
Follows https://github.com/whatwg/html/pull/7361.

@sefeng211, feel free to either review these, or just incorporate them into your local Gecko checkout and take over the testing that way. It's hard to know if I got things right and I'm sure you'd be in a better position as the one implementing :). But I wanted to help out by giving more coverage, if I could.

### CODE REVIEW

File: html/semantics/interactive-elements/the-dialog-element/dialog-focus-shadow.html [hunk 1]
```diff
@@ -1,62 +1,224 @@
...
   </template>
   <button tabindex="-1" class="focus-me">Focusable</button>
 </dialog>
 
 <script>
 for (const template of document.querySelectorAll(".turn-into-shadow-tree")) {
   const div = document.createElement("div");
-  div.attachShadow({ mode: "open" });
+  div.attachShadow({ mode: "open", delegatesFocus: template.classList.contains("delegates-focus")  });
+  div.toggleAttribute("autofocus", template.classList.contains("autofocus"));
```
Comment (nt1m) [inline html/semantics/interactive-elements/the-dialog-element/dialog-focus-shadow.html:210]:
```suggestion
  div.autofocus = template.classList.contains("autofocus");
```

**Comment by domenic** [comment]
@nt1m @emilio @annevk review appreciated, as I believe https://github.com/whatwg/html/pull/7361 is about ready to merge.

**Comment by nt1m** [comment]
cc @sefeng211

**Comment by domenic** [comment]
Another ping for reviews so we can land the spec change... maybe @josepharhar? :)

**Review by emilio — APPROVED** [review]
LGTM

**Pull request merged**